src/serialization.rs

```diff
@@ -4,15 +4,17 @@ use serde::ser::Serialize;
 use serde_json::{from_str, to_string, Value};
 use serde_json::map::Map;
 
-use errors::{Result};
+use errors::Result;
 use header::Header;
 
 
 /// The return type of a successful call to decode
 #[derive(Debug)]
-pub struct TokenData<T: Deserialize> {
+pub struct TokenData<T>
+    where for<'a> T: Deserialize<'a>
```

Comment:
I filed https://github.com/Manishearth/rust-clippy/issues/1689 to catch this case with a warning in Clippy. Apparently even the standard library does it wrong sometimes!